src/components/implementation/tests/micro_booter/micro_booter.c

```diff
@@ -97,6 +143,66 @@ test_mem(void)
 
 volatile arcvcap_t rcc_global, rcp_global;
 volatile asndcap_t scp_global;
+long long total_asnd_cycles = 0LL, total_arcv_cycles = 0LL;
+long long start_asnd_cycles = 0LL, end_asnd_st_arcv_cycles = 0LL, end_arcv_cycles = 0LL;
+int async_test_flag = 0;
+
+static void
+async_thd_fn_perf(void *thdcap)
+{
+	thdcap_t tc = (thdcap_t)thdcap;
+	arcvcap_t rc = rcc_global;
+	thdid_t tid;
+	int rcving;
+	cycles_t cycles;
+
+	cos_rcv(rc, &tid, &rcving, &cycles);
+	/* printc("receiving thread!!\n"); */
+
+	long long iters = 0LL;
+	for (; iters < ITERATIONS + 1; iters ++) {
+		/* rdtscll(end_asnd_st_arcv_cycles); */
+		cos_rcv(rc, &tid, &rcving, &cycles);
+	}
+
+	cos_thd_switch(tc);
+}
+
+static void
+async_thd_parent_perf(void *thdcap)
+{
+	thdcap_t tc = (thdcap_t)thdcap;
+	arcvcap_t rc = rcp_global;
+	asndcap_t sc = scp_global;
+
+	cos_asnd(sc);
+	/* printc("sending thread!!\n"); */
+
+	start_asnd_cycles = 0LL, end_asnd_st_arcv_cycles = 0LL, end_arcv_cycles = 0LL;
+	/* simplest measurement: not accurate. */
+	long long iters = 0LL;
+	rdtscll(start_asnd_cycles);
+	for (; iters < ITERATIONS; iters ++) {
+		/* start_asnd_cycles = 0LL, end_asnd_st_arcv_cycles = 0LL, end_arcv_cycles = 0LL;
+		rdtscll(start_asnd_cycles); */
+		cos_asnd(sc);
+		/* rdtscll(end_arcv_cycles);
+		total_asnd_cycles += (end_asnd_st_arcv_cycles - start_asnd_cycles);
+		total_arcv_cycles += (end_arcv_cycles - end_asnd_st_arcv_cycles); */
+	}
+	rdtscll(end_arcv_cycles);
+	total_asnd_cycles = (end_arcv_cycles - start_asnd_cycles) / 2ll ;
+	printc("Average ASND/ARCV (Total: %lld / Iterations: %lld ): %lld\n", 
+		total_asnd_cycles, (long long) (ITERATIONS), (total_asnd_cycles / (long long)(ITERATIONS)));
+	/* printc("Average ARCV (Total: %lld / Iterations: %lld ): %lld\n", 
```

Comment:
This file is as much of documentation of how to use the API as it is a microbenchmark.  Thus we need it to be unambiguous and clear.